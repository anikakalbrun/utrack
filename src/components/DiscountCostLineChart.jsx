import React from 'react';
import {object} from 'prop-types';
import '../styles/css/Chart.css';

import LegendItem from './LegendItem';
import Legend from './Legend';
import {colorLine1, colorLine2} from '../chartFixtures';

const LineChart = require("react-chartjs").Line;

const chartOptions = {
    bezierCurve: false,
    pointDot: true,
    pointDotRadius: 4,
    pointDotStrokeWidth: 2,
    pointHitDetectionRadius: 10,
    responsive: true,
    maintainAspectRatio: false
};

DiscountCostLineChart.propTypes = {
    data: object
};

export default function DiscountCostLineChart({data}) {
    return (
        <div className='width100 p-50 h-300'>
            <Legend>
                <LegendItem color={colorLine2} label='Monthly'/>
                <LegendItem color={colorLine1} label='Cumulative'/>
            </Legend>
            {data && data.datasets && <LineChart data={data}
                                options={chartOptions}
                                redraw/>
            }
        </div>
    );
}

//TODO: upgrade from react-chartjs to react-chartjs-2
// adding tooltips option to chartOptions is being ignored => I can not modify the tooltip label color
// to fix this issue I could upgrade to react-chartjs-2
// but I can not upgrade because when using react-chartjs-2 I get an compilation error that is still an open issue on github:
// https://github.com/jerairrest/react-chartjs-2/issues/233

//upgrading should also remove the warning:
//[Violation] Added non-passive event listener to a scroll-blocking <some> event. Consider marking event handler as 'passive' to make the page more responsive. See <URL>