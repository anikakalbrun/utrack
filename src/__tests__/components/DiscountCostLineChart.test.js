import React from 'react';
import DiscountCostLineChart from '../../components/DiscountCostLineChart';

let wrapper;
const data = {
    labels: ["January", "February"],
    datasets: []
};

describe('DiscountCostLineChart', () => {

    it('renders Legend and two LegendItem components', () => {
        wrapper = shallow(<DiscountCostLineChart/>);
        expect(wrapper.props().children).toMatchSnapshot();
    });

    it('renders LineChart when data prop is passed', () => {
        wrapper = shallow(<DiscountCostLineChart data={data}/>);
        expect(wrapper.props().children).toMatchSnapshot();
    });

    it('renders the container div', () => {
        wrapper = shallow(<DiscountCostLineChart data={data}/>);
        expect(wrapper.find('div.width100.p-50.h-300').length).toEqual(1);
    });

});
