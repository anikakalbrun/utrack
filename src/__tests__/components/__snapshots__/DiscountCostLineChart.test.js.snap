// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountCostLineChart renders Legend and two LegendItem components 1`] = `
Array [
  <Legend>
    <LegendItem
        color="rgba(151,187,205,1)"
        label="Monthly"
    />
    <LegendItem
        color="rgba(220,220,220,1)"
        label="Cumulative"
    />
</Legend>,
  undefined,
]
`;

exports[`DiscountCostLineChart renders LineChart when data prop is passed 1`] = `
Array [
  <Legend>
    <LegendItem
        color="rgba(151,187,205,1)"
        label="Monthly"
    />
    <LegendItem
        color="rgba(220,220,220,1)"
        label="Cumulative"
    />
</Legend>,
  <LineChart
    data={
        Object {
            "datasets": Array [],
            "labels": Array [
              "January",
              "February",
            ],
          }
    }
    options={
        Object {
            "bezierCurve": false,
            "maintainAspectRatio": false,
            "pointDot": true,
            "pointDotRadius": 4,
            "pointDotStrokeWidth": 2,
            "pointHitDetectionRadius": 10,
            "responsive": true,
          }
    }
    redraw={true}
/>,
]
`;
